pkg/localstorev2/recover.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storer
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal/chunkstore"
+	"github.com/ethersphere/bee/pkg/sharky"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	sharkyDirtyFileName = ".DIRTY"
+)
+
+func sharkyRecovery(sharkyBasePath string, store storage.Store, opts *Options) (closerFn, error) {
```

Comment:
Since this could be a potentially long-running task, we should be taking in a context in the New function and handling shutdown.